Object store item size fix

darkwatchuk opened the pull request:
ObjectStore ObjectMetaData.Size is currently an int and causes a JSON deserialization crash when pointing to an object store with large files. 

Changing to a ulong fixes this issue.

e.g. someone has uploaded large files to this object store on demo.nats.io
```csharp
   [Fact]
   public async Task LargeDemoTest()
   {
       var cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
       var cancellationToken = cts.Token;

       await using var nats = new NatsConnection(new NatsOpts { Url = "nats://demo.nats.io" });
       var js = new NatsJSContext(nats);
       var ob = new NatsObjContext(js);

       var objStore = await ob.GetObjectStoreAsync("omarchy-3_0_1", cancellationToken);

       await foreach (var thing in objStore.ListAsync(cancellationToken: cancellationToken))
       {
           Console.WriteLine($"{thing.Name} : {thing.Size}");
       }
   }
```

darkwatchuk committed:
Make the parameter-less StreamConfig constructor public

darkwatchuk committed:
Merge branch 'nats-io:main' into main

darkwatchuk committed:
Merge branch 'nats-io:main' into main

darkwatchuk committed:
Merge branch 'nats-io:main' into main

darkwatchuk committed:
Make NatsMsg<T>.Build Public

mtmk committed:
Add docs

mtmk committed:
Add docs

[no ci]

darkwatchuk committed:
Merge branch 'nats-io:main' into main

darkwatchuk committed:
Merge branch 'nats-io:main' into main

darkwatchuk committed:
Change ObjectStore ObjectMetadata.Size to ulong from int

mtmk committed:
Copilot feedback

mtmk commented:
Aligning with ADR-20:

https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-20.md#objectinfo
```go
type ObjectInfo struct {
    ObjectMeta
    Bucket  string    `json:"bucket"`
    NUID    string    `json:"nuid"`

    // the total object size in bytes
    Size    uint64    `json:"size"`    // <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<

    ModTime time.Time `json:"mtime"`

    // the total number of chunks
    Chunks  uint32    `json:"chunks"` // <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<

    // as in http, <digest-algorithm>=<digest-value>
    Digest  string    `json:"digest,omitempty"`
    Deleted bool      `json:"deleted,omitempty"`
}
```

mtmk committed:
Use `uint` for `Chunks` field and update related comments/test cases.

mtmk committed:
Add test for validating metadata field types against NATS spec

mtmk commented on tests/NATS.Client.ObjectStore.Tests/ObjectStoreTest.cs:
```diff
@@ -622,4 +622,39 @@ await Assert.ThrowsAsync<NatsObjNotFoundException>(async () =>
+        var info = await store.GetInfoAsync("k1", cancellationToken: cancellationToken);
+
+        // Verify Size is ulong
+        ulong size = info.Size; // Should compile without error
```
since using large files won't be possible I just check the type here.

mtmk commented on src/NATS.Client.ObjectStore/Models/ObjectMetadata.cs:
```diff
@@ -53,7 +54,7 @@ public record ObjectMetadata
     [JsonPropertyName("chunks")]
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
-    public int Chunks { get; set; }
+    public uint Chunks { get; set; }
```
turns out this one is not align with the spec either (ADR-20)

scottf replied:
I have long (int 64) in both java and .net v1. ADR has uint32 so I think this is fine.

mtmk reviewed (approved):
LGTM thanks @darkwatchuk 

(I shall ask for one more review since I added a change too)

mtmk committed:
Keep format happy

Disable style warnings in ObjectStoreTest to ensure compatibility with type validation tests

darkwatchuk commented:
> LGTM thanks @darkwatchuk
> 
> (I shall ask for one more review since I added a change too)

Thank you!

mtmk commented:
sorry for the delay @darkwatchuk we should get this merged soon then I'm hoping to get it released some time this week.

scottf reviewed (approved):
LGTM
